### ISSUE TITLE
Set homepage as Blog (broken with beta 16 update)

### ISSUE THREAD

**Issue by IamDarkle** [v17development/flarum-blog#85]
With the latest version of the extension the configuration of the homepage as Blog is broken, it picks up discussions instead of blog category posts. 

To see the live bug you can go to [flarumes](https://flarum.es) or [brickdose](https://brickdose.com). To see the posts that are actually from the blog category add /blog to the url (it's a bit weird, when you go to e.g. [brickdose.com/blog](https://brickdose.com/blog) it shows the correct blog posts but automatically deletes /blog from the url) **Edit**: this cannot be done as a guest user (it gives you error) only if you are logged in, I have also temporarily removed the blog as a homepage on flarum.es because of the bug.

With /all in the url ([brickdose.com/all](https://brickdose.com/all)) you can see that it is actually taking posts from discussions (with categories not added to the blog extension).

After a lot of testing I get no errors in console or logs. I rule out that this is an incompatibility with other extensions, the error occurred equally with about 60 extensions as with 4 extensions. 

I leave some details of the server in case they are useful, anyway I could replicate the error on a xampp server (Windows) so I do not know how useful they will be...

```
Operating System: CentOS Linux 8
Kernel: Linux 4.18.0-240.15.1.el8_3.x86_64
Flarum core 0.1.0-beta.16
nginx version: nginx/1.18.0
10.3.28-MariaDB
PHP version: 8.0.3
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, readline, Reflection, SPL, session, standard, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, intl, json, exif, mysqlnd, PDO, Phar, SimpleXML, sockets, sodium, sqlite3, tokenizer, xml, xmlwriter, xsl, mysqli, pdo_mysql, pdo_sqlite, xmlreader, Zend OPcache
```

When I first got the error the server is running PHP 7.4.16 I decided to upgrade to PHP 8.0.3 for that was the problem, but the issue persists in the same way.

**Comment by ctml91** [comment]
I noticed the same during my test today. Unfortunately for me it renders it unusable in the meantime :(

**Comment by jaspervriends** [comment]
Thanks for reporting, I'll look at it this weekend

**Comment by IamDarkle** [comment]
> Thanks for reporting, I'll look at it this weekend

Thank you Jasper! I've edited the initial post with some more information after the tests I did yesterday, I hope it's helpful 😀

**Comment by ihhddis** [comment]
I had the same problem when I upgraded to beta 16, I went back to beta 15 which works, I don't know what happened with the upgrade

**Comment by jaspervriends** [comment]
Hi guys, after investigating the issue for a while now (and breaking my mind about it as it used to work before without any issues), it came to my notice that this is a bug in the Flarum core of beta-16. The Flarum team appears to have already fixed the bug, so it the will become available in the next version of flarum...

More information about the issue here: https://github.com/flarum/core/issues/2752#issuecomment-813077937

There isn't much we can do about this, but I'm still exploring some options... If anyone has some feedback/ideas, please let me know

**Comment by ctml91** [comment]
Wow thanks Jasper! I'm sorry to waste your time, of course our first instinct is to start looking at the extension.

What is odd is that users on discuss report blog as homepage working. I'm not sure why it would differ for people.

Anyways, looks like you cracked the case again 👍  At least I now know that my upgrade to beta 16 will not be happening so I can stop worrying about and wait for the next RC or stable :)

**Comment by IamDarkle** [comment]
> There isn't much we can do about this, but I'm still exploring some options... If anyone has some feedback/ideas, please let me know

Thank you very much for your dedication, it seemed such a weird bug that I really thought it was my server's fault. 

With the possible breaking changes in beta 17-1.0 it's possible that I'll be on beta 16 for a while, in the issue the user says he has patched the bug by applying the commints, is it possible to do that in production from the server? Do you think it's too complicated or impossible to do that in my case?

**Comment by jaspervriends** [comment]
Fortunately this has automatically been fixed in the new Flarum v1.0.0 version 🎉

I'll close this issue :)
